_posts/2024/2024-02-06-our-process.md

```diff
@@ -0,0 +1,453 @@
+before diving into implementation.
+The goal here is *not* to create a pile of UML diagrams that no one will ever look at again,
+but rather to explain what the problem actually is,
+what factors a solution must take into account,
+what alternatives there are,
+and what we're *not* going to do.
+Some designs go through two or three overhauls before work starts;
+in almost every case,
+checking our thinking like this has saved us pain downstream.
+
+-   Summary: Weekly sync to present and discuss technical proposals
+-   Key Questions:
+    -   What problem is the design solving?
+    -   For who?
+    -   How are they solving the problem today?
+    -   Why isn't that good enough?
+    -   What is out of scope?
+-   Frequency: weekly
+-   Responsible: VP Engineering
+-   Contributors: entire engineering team
+-   Products:
+    -   Designs for new features or applications
+    -   Lateral awareness of who's doing what and why
+    -   Buy-in from the team (or at least a chance to air our differences)
+
+> A recent change is to update the design doc a few weeks after work starts
+> to explain where and why reality has diverged from the plan.
+> Doing this extends the useful life of that document,
+> but more importantly,
+> it's a chance to reflect on what the code is teaching us
+> and share that with other people
+> so that our next design(s) can be better.
+
+### Weekly Friday Engineering Demos
+
+One of our more experienced team members instituted Friday afternoon demo sessions last year,
+and we have all become believers.
+Some demos show off a new feature;
+others walk the audience through a bug that has been fixed
+or introduce the team to useful third-party tools.
+Demos runs anywhere from a few minutes to half an hour,
+with most being near the low end of that range.
+It's a fun way to wrap up the week and remind ourselves that we're actually making progress.
+
+-   Summary: Weekly sync for team to share ongoing work and discoveries
+-   Key Questions:
+    -   It does *what*??
+-   Frequency: weekly
+-   Responsible: volunteer engineer
+-   Contributors: entire engineering team
+-   Products:
+    -   Team spirit
+    -   Lateral knowledge transfer
+
+### User Support
+
+As mentioned earlier,
+we now take week-long turns monitoring the `#eng-support` channel
+and responding to users' problems.
+In some cases (e.g., when it's my turn to do support)
+the response usually takes the form of,
+"OK, let me pull in someone who actually understands what's going on here,"
+but whether the on-call engineer handles the problem themself or hands it off,
+it gives us all regular reminders of how our software is being used
+and what state it is actually in.
+
+-   Summary: Ongoing support of issues and requests
+-   Frequency: ongoing
+-   Responsible: designated engineer (rotation)
+-   Contributors: team members as needed
+-   Products:
+    -   Happier users
+    -   Issues in the issue tracker
```

Comment:
I was going to say that Ideas for future product improvements was another product, but maybe that's under "happier users"?

Reply:
expanded